controllers/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -255,6 +257,56 @@ func (r *Reconciler) updateStatus(ctx context.Context, tmplTier *toolchainv1alph
 	return r.Client.Status().Update(ctx, tmplTier)
 }
 
+func (r *Reconciler) handleDeletion(ctx context.Context, tier *toolchainv1alpha1.NSTemplateTier) error {
+	list := &toolchainv1alpha1.SpaceList{}
```

Comment:
Just curious, What if we check first if there is already a finalizer present or not , then can we directly return ?

Reply:
Okay , so in the current logic, i see it checks `util.IsBeingDeleted(&s)` ,  would checking `util.HasFinalizer` would be before listing spaces , am i understanding this correctly?

Reply:
I mean that we filter out those that are being deleted

Reply:
I was actually curious about checking a generic finalizer before listing spaces.

Reply:
as for your question Feny (I misunderstood it before), we don't need to make yet another check because if the resource is being deleted and it doesn't have any finalizer, then it should be removed from the etcd, which means that we shouldn't get any other reconciles.

Reply:
Thanks for Explaining !!